docs/component-stability.md

```diff
@@ -66,6 +66,74 @@ Stable components MUST be compatible between minor versions unless critical secu
 component owner MUST provide a migration path and a reasonable time frame for users to upgrade. The same rules from beta
 components apply to stable when it comes to configuration changes.
 
+#### Observability requirements
+
+Stable components should emit enough internal telemetry to let users detect errors, as well as data
+loss and performance issues inside the component, and to help diagnose them if possible.
+
+The internal telemetry of a stable component should allow observing the following:
```

Comment:
The text seems to exclude extensions, I would clarify it like this:
```suggestion
The internal telemetry of a stable pipeline component should allow observing the following:
```

Reply:
Discussed offline, we can probably add some wording around being able to observe performance and errors but it will be quite generic